planning/obstacle_cruise_planner/src/polygon_utils.cpp

```diff
@@ -233,16 +233,16 @@ geometry_msgs::msg::PointStamped calcNearestCollisionPoint(
                              traj_front_pose, vehicle_max_longitudinal_offset, 0.0, 0.0)
                              .position;
     if (is_driving_forward) {
-      segment_points.at(0) = traj_front_pose.position;
-      segment_points.at(1) = front_pos;
+      segment_points.push_back(traj_front_pose.position);
+      segment_points.push_back(front_pos);
```

Comment:
Sorry for the confusion, It's ok to use `.at` if making by `std::vector<geometry_msgs::msg::Point> segment_points(2);` 
I meant that if maingit by `std::vector<geometry_msgs::msg::Point> segment_points;` you need tp use `push_back`